.github/workflows/build.yaml

```diff
@@ -28,7 +28,7 @@ concurrency:
 jobs:
   cpp-build:
     secrets: inherit
-    uses: rapidsai/shared-workflows/.github/workflows/conda-cpp-build.yaml@branch-24.10
+    uses: rapidsai/shared-workflows/.github/workflows/conda-cpp-build.yaml@python-3.12
```

Comment:
just curious if this is a new naming convention (going from RAPIDS branch name to python version) or just a temporary change for python 3.12

Reply:
Fair question! Nope this is only temporary.

The process I'm using is like this:

* put up CI changes on a new `shared-workflows` branch: https://github.com/rapidsai/shared-workflows/pull/213
* migrate all the repos to using those new CI configs one-by-one (with PRs like this one)
* once everything is migrated, merge those changes into `shared-workflows`
* change all the repos back to referencing a `branch-{YY}.{MM}` branch

This way, we can do the work incrementally. Instead of, say, adding Python 3.12 jobs that then all RAPIDS repos have to immediately become compliant with.